### PR TITLE
Added a Directory sync component

### PR DESCRIPTION
**Pull request by DavidCroftDKFZ**
* Added new container to bbmri/docker-compose.yml.
* Added set up documentation to README.

### CODE REVIEW

File: bbmri/docker-compose.yml [hunk 1]
```diff
@@ -104,6 +104,17 @@ services:
       - /etc/bridgehead/trusted-ca-certs:/conf/trusted-ca-certs:ro
       - ./root.crt.pem:/conf/root.crt.pem:ro
 
+  directory_sync_service:
+    image: "samply/directory_sync_service"
+    environment:
+      DS_DIRECTORY_URL: ${DS_DIRECTORY_URL}
+      DS_DIRECTORY_USER_NAME: ${DS_DIRECTORY_USER_NAME}
+      DS_DIRECTORY_PASS_CODE: ${DS_DIRECTORY_PASS_CODE}
+      DS_FHIR_STORE_URL: "http://bridgehead-bbmri-blaze:8080"
```
Comment (patrickskowronekdkfz) [inline bbmri/docker-compose.yml:113]: Should be default in the spring application.yml

File: bbmri/docker-compose.yml [hunk 2]
```diff
@@ -104,6 +104,17 @@ services:
...
+      DS_TIMER_CRON: ${DS_TIMER_CRON}
+      DS_RETRY_MAX: ${DS_RETRY_MAX}
```
Comment (patrickskowronekdkfz) [inline bbmri/docker-compose.yml:115]: Should be default in the spring application.yml

File: bbmri/docker-compose.yml [hunk 3]
```diff
@@ -104,6 +104,17 @@ services:
...
+      DS_RETRY_INTERVAL: ${DS_RETRY_INTERVAL}
```
Comment (patrickskowronekdkfz) [inline bbmri/docker-compose.yml:116]: Should be default in the spring application.yml

File: README.md [hunk 4]
```diff
@@ -131,6 +131,41 @@ All of the Bridgehead's outgoing connections are secured by transport encryption
 
 Your Bridgehead's actual data is not stored in the above directories, but in named docker volumes, see `docker volume ls` and `docker volume inspect <volume_name>`.
 
+### Directory sync
+
+This is an optional feature for bbmri projects.  It keeps the [BBMRI Directory](https://directory.bbmri-eric.eu/) up to date with the number of samples, etc.  kept in your biobank. It also updates the local FHIR store with the latest contact details etc. from the Directory.  You must explicitly enable this feature if you want to make use of it.
```
Comment (patrickskowronekdkfz) [inline README.md:136]:
```suggestion
This is an optional feature for bbmri projects. It keeps the [BBMRI Directory](https://directory.bbmri-eric.eu/) up to date with your local data eg. number of samples. It also updates the local FHIR store with the latest contact details etc. from the Directory.  You must explicitly set your country specific directory url, username and password to enable this feature.
```

File: README.md [hunk 5]
```diff
@@ -131,6 +131,41 @@ All of the Bridgehead's outgoing connections are secured by transport encryption
...
+
+Full details can be found in [directory_sync_service](https://github.com/samply/directory_sync_service).
+
+To enable it, you will need to add some extra variables to the ```bbmri.conf``` file in your GitLab repository, for example:
```
Comment (patrickskowronekdkfz) [inline README.md:140]:
```suggestion
To enable it, you will need to set these variables in the ```bbmri.conf``` file of your GitLab repository. Here is an example config:
```

File: README.md [hunk 6]
```diff
@@ -131,6 +131,41 @@ All of the Bridgehead's outgoing connections are secured by transport encryption
...
+```
+### Directory sync service
+DS_DIRECTORY_URL=https://directory.bbmri-eric.eu
+DS_DIRECTORY_USER_NAME=your_directory_username
+DS_DIRECTORY_USER_PASS=qwdnqwswdvqHBVGFR9887
+DS_TIMER_CRON="0 22 * * *"
+```
+You must contact the Directory for your national node to find the URL, and to register as a user.
+
+Additionally, you should choose when you want Directory sync to run. In the example above, this is set to happen at 10 pm every evening. You can modify this to suit your requirements. The timer specification should follow the cron convention.
```
Comment (patrickskowronekdkfz) [inline README.md:151]:
```suggestion
Additionally, you should choose when you want Directory sync to run. In the example above, this is set to happen at 10 pm every evening. You can modify this to suit your requirements. The timer specification should follow the [cron](https://crontab.guru) convention
```

File: README.md [hunk 7]
```diff
@@ -131,6 +131,41 @@ All of the Bridgehead's outgoing connections are secured by transport encryption
...
+
+Once you have made the changes, update your local configuration:
```
Comment (patrickskowronekdkfz) [inline README.md:153]:
```suggestion
Once you edited the gitlab config. The bridgehead will autoupdate the config with the values and will sync the data.
```

File: README.md [hunk 8]
```diff
@@ -131,6 +131,41 @@ All of the Bridgehead's outgoing connections are secured by transport encryption
...
+cd /etc/bridgehead
+sudo git pull
+sudo chown -R bridgehead * .git*
+```
+
+Then restart the Bridgehead:
+
+```shell
+sudo systemctl restart bridgehead@bbmri.service
+```
```
Comment (patrickskowronekdkfz) [inline README.md:165]: Not needed

**Review by patrickskowronekdkfz — CHANGES_REQUESTED** [review]

**Comment by patrickskowronekdkfz** [comment]
- [ ] TODO: add these values to the template and each repo

**Review by patrickskowronekdkfz — APPROVED** [review]
Needs a second reviewer

**Comment by patrickskowronekdkfz** [comment]
@torbrenner please look at this PR

File: bbmri/directory-sync-compose.yml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+services:
+  directory_sync_service:
+    image: "samply/directory_sync_service"
```
Comment (torbrenner) [inline bbmri/directory-sync-compose.yml:3]: Please use `docker.verbis.dkfz.de/cache/samply/directory_sync_service`

**Review by torbrenner — CHANGES_REQUESTED** [review]

File: bbmri/directory-sync-compose.yml (hunk 9 above)
Comment (patrickskowronekdkfz) [reply bbmri/directory-sync-compose.yml]: @torbrenner done

**Review by torbrenner — APPROVED** [review]

**Pull request merged**